src/redux/features/asyncUtils.ts

```diff
@@ -7,17 +7,32 @@ export type StatusT<T> = { status: Statuses; data?: T };
 export type Status = StatusT<unknown>; // when data is not relevant in the response
 
 export const buildReducers =
-  <T>(asyncThunk: AsyncThunk<any, any, any>, useArgAsKey?: boolean) =>
+  <T>(asyncThunk: AsyncThunk<any, any, any>, useArgAsKey?: boolean, compositeKeyFn?: (args: unknown) => string) =>
   (builder: CaseReducers<any, any> | ActionReducerMapBuilder<any>) =>
     (builder as any)
-      .addCase(asyncThunk.pending, setStatus<T>('pending', useArgAsKey))
-      .addCase(asyncThunk.fulfilled, setStatus<T>('success', useArgAsKey))
-      .addCase(asyncThunk.rejected, setStatus<T>('error', useArgAsKey));
+      .addCase(asyncThunk.pending, setStatus<T>('pending', useArgAsKey, compositeKeyFn))
+      .addCase(asyncThunk.fulfilled, setStatus<T>('success', useArgAsKey, compositeKeyFn))
+      .addCase(asyncThunk.rejected, setStatus<T>('error', useArgAsKey, compositeKeyFn));
 
 export const setStatus =
-  <T>(status: Statuses, useArgAsKey?: boolean) =>
+  <T>(status: Statuses, useArgAsKey?: boolean, compositeKeyFn?: (arg: unknown) => string) =>
   (state: any, action: any) => {
-    const key = useArgAsKey ? action.meta.arg : action.meta.requestId;
     const data: T = action.payload as T;
+
+    state[action.meta.requestId] = { status, data };
```

Comment:
This is another change, previously it was only storing at the given key if `useArgAsKey` was used. With this change it is stored in both places, `requestId` and the arg / composite key. This is on purpose so consumers that care about a particular request ID can easily track the status and consumers that only care about the underlying data can access it via composite key or arg key. Let me know if you think that's alright or if I should go back to only setting it in one location in state.

Reply:
Ah got it, missed this comment.
This is fine, it does increase the store size which may not be ideal if there is no use-case. Thoughts on adding this later once we identify a need?

Reply:
Sounds good to me

Reply:
Ignore my earlier comment, we crossed wires for a bit.